neural_speed/application/main_pybind.cpp

```diff
@@ -54,6 +56,241 @@
 
 namespace py = pybind11;
 
+namespace {
+struct Query {
+  uint64_t id;
+  std::vector<model_vocab::id> token_ids;
+  Query() {}
+  Query(uint64_t id, const pybind11::array_t<model_vocab::id, py::array::c_style | py::array::forcecast>& token_ids)
+      : id(id), token_ids(token_ids.data(), token_ids.data() + token_ids.size()) {
+    assert(token_ids.ndim() == 1 || (token_ids.ndim() == 2 && token_ids.shape(0) == 1));
+  }
+
+  std::string to_string() const {
+    const std::string repr_ids(py::str(py::array_t<int, py::array::c_style>(token_ids.size(), token_ids.data())));
+    return std::to_string(id) + ": " + repr_ids;
+  }
+};
+
+using Response = Query;
```

Comment:
any specific reasons?

Reply:
@DDEle

Reply:
The data structure happens to be the same. Currently, a query consists of a query ID and prompt tokens IDs, while a response consists of it responding query ID and generated token IDs. They may become different if we decide to add some fields to one but not to another.

Reply:
> The data structure happens to be the same. Currently, a query consists of a query ID and prompt tokens IDs, while a response consists of it responding query ID and generated token IDs. They may become different if we decide to add some fields to one but not to another.

add this into comments

Reply:
Added in 664630b